src/lib/class/Server/handle_clients.cpp

```diff
@@ -0,0 +1,96 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   handle_clients.cpp                                 :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: mcourtoi <mcourtoi@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/02/20 21:11:16 by mcourtoi          #+#    #+#             */
+/*   Updated: 2024/02/29 20:08:41 by mcourtoi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "IrcMessage.hpp"
+#include "class/Server.hpp"
+#include "replies.hpp"
+
+/**
+ * @brief Change the socket flag to non blocking
+ *
+ * @param socket the socket to set non blocking
+ */
+void control_socket(int const socket)
+{
+	int flags = fcntl(socket, F_GETFL, 0);
+
+	if (flags == -1)
+		throw ProblemWithFcntlFlags();
+	flags |= O_NONBLOCK;
+	if (fcntl(socket, F_SETFL, flags) == -1)
+		throw ProblemWithSettingNonBlock();
+}
+
+/**
+ * @brief call recv() to get what the client wrote in its socket
+ *
+ * @param client the client to get the message from
+ */
+void Server::rcv_client_event(Client *client)
+{
+	client->set_time_last_msg();
+
+	char    buffer[4096];
+	ssize_t bytes_read;
+
+	bytes_read = recv(client->get_socket(), buffer, 4096, 0);
+	if (bytes_read == -1)
+		throw ProblemWithRecv();
+	if (bytes_read == 0)
+	{
+		client->disconnect();
+		this->remove_client(client);
+		return;
+	}
+	buffer[bytes_read] = '\0';
```

Comment:
Out-of-bounds error here. Easy segfault.

Reply:
![image](https://github.com/Saphirac/ft_irc/assets/89741910/dadc052d-07f9-48d0-8fa1-96da23e9ce3b)

corrected it as such